src/Microsoft.Health.Core/Features/Health/AsyncData.cs

```diff
@@ -0,0 +1,38 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+
+namespace Microsoft.Health.Core.Features.Health;
+
+/// <summary>
+/// Asynchronously waits for the data to be set before the data can be retrieved
+/// </summary>
+/// <typeparam name="T">The data type</typeparam>
+public class AsyncData<T> where T : class
```

Comment:
Renamed this to AsyncData to not conflict with https://github.com/microsoft/dicom-server/blob/0d916dd9f5d8c6b27c0b434dbb9ef1ac64283d0e/src/Microsoft.Health.Dicom.Core/Features/Common/AsyncCache.cs#L13

Reply:
Maybe `Cache<T>`? `ResultCache<T>`? `ValueCache<T>`? `SingletonCache<T>`?

Reply:
Renamed to `ValueCache<T>`